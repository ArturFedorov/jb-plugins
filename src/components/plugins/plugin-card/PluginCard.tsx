import React, { FunctionComponent, useMemo } from 'react';
import { NavLink } from 'react-router-dom';
import { StarRating } from '../../common/star-rating/StarRating';
import { PluginIcon } from '../plugin-icon/PluginIcon';
import styles from '../plugins.module.scss';
import { IPlugin } from '../../../shared/interfaces/models/IPlugin';
import dateFormat from '../../../shared/utils/date-util/date.util';
import { formatNumber } from '../../../shared/utils/format-util/format.util';
import pluginIcon from '../../../assets/icons/jetbrains-logo.svg';

export interface IPluginCardProps {
  plugin: IPlugin;
}

export const PluginCard: FunctionComponent<IPluginCardProps> = ({ plugin }) => {
  const formattedDate = useMemo(() => dateFormat(plugin.date), [plugin.date]);
  const memoizedStarRating = useMemo(() => <StarRating rating={plugin.rating} />, [plugin.rating]);

  return (
    <NavLink to={`/plugins/${plugin.id}`} className={styles.pluginCardLink}>
      <div className={styles.pluginCard}>
        <div className={styles.pluginCardHeader}>
          <PluginIcon iconUrl={plugin.icon || pluginIcon} className={styles.pluginCardIcon} />
          <div className={styles.pluginCardSection}>
            <h3 className="is-lighter marginless">{plugin.name}</h3>
            {memoizedStarRating}
          </div>
        </div>
        <div className={styles.pluginCardDescription}>
          <span>{plugin.description}</span>
        </div>
        <div className={styles.pluginCardFooter}>
          <span className="is-caption">{formatNumber(plugin.downloads)} downloads</span>
          <span className="is-caption">{formattedDate}</span>
        </div>
      </div>
    </NavLink>
  );
};
