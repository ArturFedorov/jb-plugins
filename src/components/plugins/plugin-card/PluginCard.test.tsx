import React from 'react';
import { shallow } from 'enzyme';
import { PluginCard } from './PluginCard';
import { TestUtil } from '../../../shared/utils/test.util';

describe('Plugin card', () => {
  const plugin = TestUtil.mockPlugin(1);

  it('should match a snapshot', () => {
    const component = shallow(<PluginCard plugin={plugin} />);
    expect(component).toMatchSnapshot();
  });
});
