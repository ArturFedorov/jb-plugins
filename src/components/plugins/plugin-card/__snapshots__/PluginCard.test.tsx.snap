// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin card should match a snapshot 1`] = `ShallowWrapper {}`;
